song-server/src/main/java/org/icgc/dcc/song/server/service/AnalysisService.java

```diff
@@ -107,9 +132,13 @@ void addFile(String id, String fileId) {
 
   public Analysis read(String id) {
     val analysis = repository.read(id);
+    if (analysis == null) {
+      return null;
+    }
```

Comment:
it seems like this pattern of returning nulls is reoccurring. Create a ticket for now regarding refactoring null-return statements, and we can review later whether its worth the refactoring. This article explains my reasoning behind null logic: https://dzone.com/articles/java-optional-objects, as well as this one https://github.com/google/guava/wiki/UsingAndAvoidingNullExplained